app-server/app/src/main/scala/user/User.scala

```diff
@@ -51,30 +74,55 @@ object UserService {
           case Order.Asc => applySort(query.sortBy(_.id.asc), sortMap.tail)
           case Order.Desc => applySort(query.sortBy(_.id.desc), sortMap.tail)
         }
-      case Some(("organizationId", order)) =>
-        order match {
-          case Order.Asc => applySort(query.sortBy(_.organizationId.asc), sortMap.tail)
-          case Order.Desc => applySort(query.sortBy(_.organizationId.desc), sortMap.tail)
-        }
       case Some((_, order)) => applySort(query, sortMap.tail)
       case _ => query
     }
   }
 
+  def usersRoleOrgJoin(query: Query[Users, UsersRow, Seq])
+                   (implicit database: Database, ec: ExecutionContext) = {
+    import database.driver.api._
+
+    val userOrgJoin =
+        query join UsersToOrganizations join Organizations on {
+        case ((user, userToOrg), org) =>
+          user.id === userToOrg.userId &&
+            userToOrg.organizationId === org.id
+      }
+
+    for {
+      ((user, userToOrg), org) <- userOrgJoin
+    } yield (user.id, org.id, org.name, userToOrg.role)
+  }
+
+  def groupUserWithOrgs(s: Seq[(String, java.util.UUID, String, String)]): Seq[UserWithOrgs] = {
```

Comment:
This function + the previous one combined are pretty opaque. I don't really know enough about the performance implications, but dealing with a sequence of tuples here is confusing.

At the very least, renaming `s` to something that is more meaningful makes sense and adding some docstrings as well